Index: linux-stage/fs/ext4/ext4.h
===================================================================
--- linux-stage.orig/fs/ext4/ext4.h
+++ linux-stage/fs/ext4/ext4.h
@@ -3033,6 +3033,9 @@ extern int ext4_sync_file(struct file *,
 extern int ext4fs_dirhash(const struct inode *dir, const char *name, int len,
 			  struct dx_hash_info *hinfo);
 
+extern int __ext4fs_dirhash(const char *name, int len,
+			    struct dx_hash_info *hinfo);
+
 /* ialloc.c */
 extern int ext4_mark_inode_used(struct super_block *sb, int ino);
 extern struct inode *__ext4_new_inode(struct user_namespace *, handle_t *,
Index: linux-stage/fs/ext4/hash.c
===================================================================
--- linux-stage.orig/fs/ext4/hash.c
+++ linux-stage/fs/ext4/hash.c
@@ -286,6 +286,7 @@ static int __ext4fs_dirhash(const struct
 	hinfo->minor_hash = minor_hash;
 	return 0;
 }
+EXPORT_SYMBOL(__ext4fs_dirhash);
 
 int ext4fs_dirhash(const struct inode *dir, const char *name, int len,
 		   struct dx_hash_info *hinfo)
